src/templates/TemplateData.ts

```diff
@@ -133,11 +141,13 @@ export async function tryGetTemplateDataFromCache(reporter: TelemetryReporter |
             const templatesMap: { [runtime: string]: Template[] } = {};
             const configMap: { [runtime: string]: Config } = {};
             for (const key of Object.keys(ProjectRuntime)) {
+                // called within loop in case setting has changed between runtimes
+                const userTemplateVersion: string | undefined = getFuncExtensionSetting(templateVersionSetting);
                 const runtime: ProjectRuntime = <ProjectRuntime>ProjectRuntime[key];
                 const feedRuntime: string = getFeedRuntime(runtime);
-                const currentRelease: string = cliFeedJson.tags[feedRuntime].release;
+                const currentRelease: string = userTemplateVersion ? userTemplateVersion : cliFeedJson.tags[feedRuntime].release;
```

Comment:
nit: Can you rename this to 'releaseVersion' like you have in tryGetTemplateDataFromCache? It might not be the 'current' version if the user specified something else